lib/google_contacts_api/group.rb

```diff
@@ -40,5 +40,25 @@ def edit_link
       _link = self["link"].find { |l| l.rel == "edit" }
       _link ? _link.href : nil
     end
+
+    def delete(api)
+      api.delete(id_path, {}, 'If-Match' => etag)
+    end
+
+    def self.create(api, title)
+      response = api.post('groups/default/full', xml_for_create(title), {}, 'Content-Type' => 'application/atom+xml')
+      Group.new(Hashie::Mash.new(JSON.parse(response.body)).entry)
+    end
+
+    def self.xml_for_create(title)
+      '<atom:entry xmlns:gd="http://schemas.google.com/g/2005" '\
```

Comment:
I'd still use a heredoc here. You can still interpolate in the heredoc, too:

http://jeff.dallien.net/posts/optional-behavior-for-ruby-heredocs